### PR TITLE
Cranelift: Add `.wat` to assembly test support and generate Wasm load/store tests for all ISAs

### PR DESCRIPTION
**Pull request by fitzgen**
This is just the `.wat`-to-assembly support from https://github.com/bytecodealliance/wasmtime/pull/5386 as well as the generated tests. By landing this first, and rebasing that on top of this, we can check for regressions and perturbations caused by moving where heap legalization is done in that PR.

@cfallin or @jameysharp want to take a look? Nothing has changed from that original PR other than minor rebase related things.

### CODE REVIEW

File: cranelift/filetests/filetests/wasm/load-store/make-load-store-tests.sh
Comment (jameysharp) [inline cranelift/filetests/filetests/wasm/load-store/make-load-store-tests.sh:3]: This comment has the wrong filename pattern in it now.

**Review by cfallin — APPROVED** [review]
LGTM!

File: cranelift/filetests/src/test_wasm.rs
Comment (jameysharp) [inline cranelift/filetests/src/test_wasm.rs:65]: I was hoping we could run legalization in the non-compile case so the wat->clif tests would show any differences in generated CLIF when you rebase the other PR on this.

**Review by jameysharp — APPROVED** [review]
On top of the benefits when the other PR is rebased, this is much easier to review by itself. Thanks! I like it. A couple possible changes below but I wouldn't block merging on them.

File: cranelift/filetests/src/test_wasm.rs
Comment (fitzgen) [reply cranelift/filetests/src/test_wasm.rs:65]: I think the wasm-to-assembly tests should catch that in practice, and I don't really want to add and enable an option that we will immediately stop using after the other PR lands.

File: cranelift/filetests/filetests/wasm/load-store/make-load-store-tests.sh
Comment (fitzgen) [reply cranelift/filetests/filetests/wasm/load-store/make-load-store-tests.sh]: Good catch!

File: cranelift/filetests/src/test_wasm.rs
Comment (jameysharp) [reply cranelift/filetests/src/test_wasm.rs:65]: Sure, that's fair.

**Pull request merged**
